### DISCUSSION TITLE
Does Wallet Connect provide the facility to which I have to notify the user (client end)

### DISCUSSION THREAD

**Discussion by rahulnrt** [WalletConnect/walletconnect-monorepo#4005] [Developer Support]
Dear team,

Dear team, I have an e-commerce application that I integrated into Wallet Connect as a crypto payment gateway. When the transaction has been completed I need to notify the user in my e-commerce application so that the user can perform further action.

Does Wallet Connect provide the facility to which I have to notify the user (client end)?

**Comment by glitch-txs** [comment]
Hi, this can be tracked certainly, which library and version are you using? could you show your dependencies? thanks!

**Comment by rahulnrt** [comment]
Sure, here it is. find the below dependencies packages I used in my
application
@walletconnect/modal-react-native,
@react-native-async-storage/async-storage react-native-get-random-values ,
react-native-modal ,
react-native-svg ,
@react-native-community/netinfo,
@walletconnect/notify-client ,
@walletconnect/react-native-compat

**Reply by glitch-txs** [reply]
notify is for wallets, if you're working on an application you should integrate Web3Inbox :)
https://docs.walletconnect.com/web3inbox/about

https://docs.walletconnect.com/web3inbox/frontend-integration/usage?platform=javascript

**Comment by rahulnrt** [comment]
will it work on react native CLI, Because I have a react native application
that I Integrate it

**Comment by rahulnrt** [comment]
This guide is being followed
 https://docs.walletconnect.com/api/notify/usage

**Comment by rahulnrt** [comment]
I have an e-commerce app, which is built on React Native. I will give you
one scenario.

Suppose the customer buys any product and the customer wants to pay using
crypto. I integrate the Wallet Connect model for crypto payment
integration. Until the customer's payment is completed. I want to show him
a loader. When the customer's payment is successfully completed I want to
show him a message "Your payment has been successfully completed"

Kindly provide a specific guide regarding this How do I show the customer a
message when its payment is completed?

**Reply by glitch-txs** [reply]
You can use Wagmi for that, please check their docs.

**Comment by rahulnrt** [comment]
Second scenario: If the user transaction failed I want to  show the user
transaction-related issue message

**Comment by rahulnrt** [comment]
can you please explain brief because I am not understand how to use
